### ISSUE TITLE
change number of training epochs when resuming training

### ISSUE THREAD

**Issue by orphan2020** [ultralytics/yolov5#3299]
## 🚀 Feature
when interrupting training, there is the nice feature to continue training. unfortunately it can only be continued to the number of epochs that where intially set. 

## Motivation

when I am not happy with the precision after a number of epochs, I would like to continue and not start all over with the training

## Pitch

possibility of changing the number of epochs, when resuming training

**Comment by glenn-jocher** [comment]
@orphan2020 YOLOv5 🚀 Learning Rate (LR) [schedulers](https://github.com/ultralytics/yolov5/blob/9f3a388cea66ca71b9bbe057929eed8fbc920a7e/train.py#L109-L142) follow predefined LR curves for the fixed number of `--epochs` defined at training start (`default=300`), and are designed to fall to a minimum LR on the final epoch for best training results. For this reason you can not modify the number of epochs once training has started.

<img width="850" alt="LR Curves" src="https://user-images.githubusercontent.com/26833433/119332195-8e113a80-bc88-11eb-8eb7-7494ff1d5c01.png">

If your training was **interrupted** for any reason you may continue where you left off using the `--resume` argument. If your training fully completed, you can start a **new** training from any model using the `--weights` argument. Examples:

## Resume Single-GPU

You may not change settings when resuming, and **no additional arguments** other than `--resume` should be passed, with an optional path to the checkpoint you'd like to resume from. If no checkpoint is passed the most recently updated `last.pt` in your `yolov5/` directory is automatically found and used:
```bash
python train.py --resume  # automatically find latest checkpoint (searches yolov5/ directory)
python train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Resume Multi-GPU

[Multi-GPU DDP trainings](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training) must be resumed with the same GPUs and DDP command, i.e. assuming 8 GPUs:
```bash
python -m torch.distributed.launch --nproc_per_node 8 train.py --resume  # resume latest checkpoint
python -m torch.distributed.launch --nproc_per_node 8 train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Start from Pretrained

If you would like to **start training** from a fully trained model, use the `--weights` argument, not the `--resume` argument:
```bash
python train.py --weights path/to/best.pt  # start from pretrained model
```

Good luck and let us know if you have any other questions!

**Comment by orphan2020** [comment]
Thanks sooooo much for clarification! This should be put in the tutorial👍

**Comment by glenn-jocher** [comment]
@orphan2020 I've updated the above response a bit to include LR curves. Thanks for the feedback!
